### PR TITLE
Add docs about multi-arch Docker builds

### PR DESCRIPTION
**Pull request by dliappis**
Enhance docker distribution project docs with prerequisites for building multi-architecture images.

### CODE REVIEW

File: distribution/docker/README.md [hunk 1]
```diff
@@ -92,6 +92,33 @@ images, and combining them with a Docker manifest. The Elasticsearch Delivery
...
+supports both `linux/amd64` **and** `linux/arm64` targets.
+
+You can verify the supported targets using `docker buildx ls`. For example, the
+following output indicates that support for `linux/arm64` is missing:
+
+```shell
+$ docker buildx ls
+NAME/NODE DRIVER/ENDPOINT STATUS  BUILDKIT PLATFORMS
+default * docker
+default default         running 20.10.21 linux/amd64, linux/386
```
Comment (pugnascotia) [inline distribution/docker/README.md:106]: I think maybe something has happened to the column alignment here? Can you check please?

File: distribution/docker/README.md [hunk 2]
```diff
@@ -130,3 +157,5 @@ Ideally this import / export stuff should be completely removed.
 [DockerTests]: ../../qa/os/src/test/java/org/elasticsearch/packaging/test/DockerTests.java
 [multi-arch]: https://www.docker.com/blog/multi-arch-build-and-images-the-simple-way/
 [ubi]: https://developers.redhat.com/products/rhel/ubi
+
+[^1]: `podman/buildah` also [support](https://github.com/containers/buildah/issues/1590) building multi-platform images.
```
Comment (pugnascotia) [inline distribution/docker/README.md:161]:
```suggestion
[^1]: `podman/buildah` also [supports building multi-platform images](https://github.com/containers/buildah/issues/1590).
```

**Review by pugnascotia — APPROVED** [review]

File: distribution/docker/README.md (hunk 1 above)
Comment (dliappis) [reply distribution/docker/README.md]: Good catch. Fixed in 47bf866e6fd7fe0ec69b26738edc8251b155e4de

**Comment by mark-vieira** [comment]
Thanks for adding this @dliappis!

**Pull request merged**